bookkeeper-server/src/main/java/org/apache/bookkeeper/client/BookKeeper.java

```diff
@@ -1471,6 +1480,121 @@ public OpenBuilder newOpenLedgerOp() {
     public DeleteBuilder newDeleteLedgerOp() {
         return new LedgerDeleteOp.DeleteBuilderImpl(this);
     }
+    
+    private static final class SyncLedgerIterator implements LedgersIterator {
+
+        private final LedgerRangeIterator iterator;
+        private final ListLedgersResultImpl parent;
+        Iterator<Long> currentRange = null;
+
+        public SyncLedgerIterator(LedgerRangeIterator iterator, ListLedgersResultImpl parent) {
+            this.iterator = iterator;
+            this.parent = parent;
+        }
+
+        @Override
+        public boolean hasNext() throws IOException {
+            parent.checkClosed();
+            if (iterator.hasNext()) {
+                return true;
+            } else if (currentRange != null) {
```

Comment:
should be move this usage of "currentRange" above, at line 1498
I mean: 
- first iterate over currentRange
- when currentRange is exhausted then you check for the main "iterator"